enhancements/update/channel-metadata.md

```diff
@@ -0,0 +1,221 @@
+[The `Accept` header][http-accept] should be set to `application/vnd.redhat.cincinnati.channels.v1+json`.
+
+The update service responds with [JSON][json], as required by [the `+json` structured syntax suffix][json-structured-syntax-suffix], with the following schema:
+
+* `channels` (required [object][json-object]), with the set of available channels.
+    Keys are channel names, like `stable-4.1`.
+    Values are [objects][json-object] with the following properties:
+
+    * `description` (optional [string][json-string]) with a paragraph of unstructured text describing the channel semantics.
+
+### In-cluster Storage
+
+This enhancement adds a new custom resource definition to `config.openshift.io`, based on the following Go types:
+
+```go
+// Channel represents an OpenShift update service channel.
+type Channel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the desired state of the channel.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ChannelSpec `json:"spec"`
+
+	// status contains information about the channel.
+	// +optional
+	Status ChannelStatus `json:"status"`
+}
+
+// ChannelSpec is the desired state of the channel.  There are no
+// properties, because the channel custom resource just reflects data
+// collected from outside the cluster.
+type ChannelSpec struct {
+}
+
+// ChannelStatus represents the status of an OpenShift update service channel.
+type ChannelStatus struct {
+  // description is a paragraph of unstructured text describing the
+	// channel semantics.
+  // +optional
+  Description string `json:"description,omitempty"`
+}
+```
+
+The cluster-version operator (CVO) will poll [the channel-list API](#channel-list-api) and populate the in-cluster `Channel` objects, which can then be consumed by other tooling, such as [`oc adm upgrade [channel]`][oc-adm-upgrade-channel] and [the web console][web-console-channel].
+This enhancement also adds a `RetrievedChannels` condition, similar to [the existing `RetrievedUpdates` condition][retrieved-updates], to alert adminstrators about failures in retrieving `Channel` objects.
+When failing to retrieve channel metadata, the CVO will leave the existing `Channel` objects untouched, under the expectation that the semantics of existing channels will evolve slowly, if at all.
+
+### User Stories
+
+#### Informed Channel Selection
+
+Alice's cluster is on release 4.5.16, and she wants to know if she can update to 4.6.
+With interfaces like the web console and `oc adm upgrade ...` exposing channel descriptions channel semantics, she will no longer have to discover [these docs][channel-docs-fast] to understand what `fast-4.6` means.
+
+By exposing these semantics via an API, it's also possible for folks using alternative update services to declare their own semantics for their own channels.
+They will not need to match patterns used by Red Hat when managing channels for our hosted update service.
```

Comment:
> `it's also possible for folks using alternative update services to declare their own semantics for their own channels.
They will not need to match patterns used by Red Hat when managing channels for our hosted update service.` 

This is a hypothesis right? Or do you know anyone doing it?

Reply:
Currently hypothetical.